From 0a73f02ed5c66c21401e6df6c49962d886171ace Mon Sep 17 00:00:00 2001
From: Basanagouda Nagappa Koppad <Basanagouda.Nagappa.Koppad@intel.com>
Date: Mon, 13 Jul 2026 03:52:00 +0000
Subject: [PATCH] Sets vendor.usb.controller based on chassis vendor.

Does a one-shot check of /sys/class/udc/.
Sets vendor.mount.ep0=1 only if UDC is already visible at that moment.

Tracked-On: NIACP3-1334
Signed-off-by: Basanagouda Nagappa Koppad <Basanagouda.Nagappa.Koppad@intel.com>

diff --git a/groups/usb-gadget/auto/auto_hal.in b/groups/usb-gadget/auto/auto_hal.in
index afe8d435..11f53678 100644
--- a/groups/usb-gadget/auto/auto_hal.in
+++ b/groups/usb-gadget/auto/auto_hal.in
@@ -11,18 +11,9 @@ esac
 update_usb_controller
 
 vendor_mount_EP0() {
-    DIR="/sys/class/udc/"
-    TIMEOUT=10
-    COUNT=0
-
-    while [ $COUNT -lt $TIMEOUT ]; do
-        if [ ! -z "$(ls -A $DIR)" ]; then
-            setprop vendor.mount.ep0 1
-            return 0
-        fi
-        sleep 1
-        COUNT=$((COUNT + 1))
-    done
+DIR="/sys/class/udc/"
+if [ ! -z $(ls -A $DIR) ]; then
+        setprop vendor.mount.ep0 1
+fi
 }
-
 vendor_mount_EP0
-- 
2.34.1

